tests/unit/reconcile/connectors/test_snowflake.py

```diff
@@ -0,0 +1,249 @@
+    It then asserts that the _get_jdbc_reader and _get_jdbc_reader_options methods of the SnowflakeDataSource instance were called with the correct arguments.
+
+    Args:
+        snowflake_data_source (fixture): A pytest fixture that returns a mock instance of the SnowflakeDataSource class.
+
+    Raises:
+        AssertionError: If the _get_jdbc_reader or _get_jdbc_reader_options methods were not called with the correct arguments.
+    """
+
+    # Create a Tables configuration object with JDBC reader options
+    table_conf = Tables(
+        source_name="supplier",
+        target_name="supplier",
+        jdbc_reader_options=JdbcReaderOptions(
+            number_partitions=100, partition_column="s_nationkey", lower_bound="0", upper_bound="100"
+        ),
+        join_columns=None,
+        select_columns=None,
+        drop_columns=None,
+        column_mapping=None,
+        transformations=None,
+        thresholds=None,
+        filters=None,
+    )
+
+    # Mock the JDBC reader object and the _get_jdbc_reader_options method
+    snowflake_data_source._get_jdbc_reader = MagicMock()
+    snowflake_data_source._get_jdbc_reader_options = MagicMock()
+
+    # Call the read_data method with the Tables configuration
+    snowflake_data_source.read_data("schema", "catalog", "select 1 from dual", table_conf)
+
+    # Assert that the _get_jdbc_reader and _get_jdbc_reader_options methods were called with the correct arguments
+    snowflake_data_source._get_jdbc_reader.assert_called_once_with(
+        "select 1 from dual", snowflake_data_source.get_jdbc_url, 'net.snowflake.client.jdbc.SnowflakeDriver'
+    )
+    snowflake_data_source._get_jdbc_reader_options.assert_called_once_with(table_conf.jdbc_reader_options)
+
+
+def test_get_schema(snowflake_data_source):
+    """
+    This test function verifies the behavior of the get_schema method of the SnowflakeDataSource class.
+
+    The function calls the get_schema method with predefined table, schema, and catalog names and does not assert any conditions.
+
+    Args:
+        snowflake_data_source (fixture): A pytest fixture that returns a mock instance of the SnowflakeDataSource class.
+    """
+
+    snowflake_data_source.get_schema("supplier", "schema", "catalog")
+
+
+def test_get_schema_query(snowflake_data_source):
+    """
+    This test function verifies the behavior of the _get_schema_query method of the SnowflakeDataSource class.
+
+    The function calls the _get_schema_query method with predefined table, schema, and catalog names and asserts that the returned SQL query matches the expected format.
+
+    Args:
+        snowflake_data_source (fixture): A pytest fixture that returns a mock instance of the SnowflakeDataSource class.
+
+    Raises:
+        AssertionError: If the returned SQL query does not match the expected format.
+    """
+
+    schema = snowflake_data_source._get_schema_query("supplier", "schema", "catalog")
```

Comment:
Tests must not access private members

Reply:
Yes, Not accessing the private members now. Doing assertion in lsql way